staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning/algorithm_test.go

```diff
@@ -66,7 +70,8 @@ func TestPrune(t *testing.T) {
 					},
 				},
 			},
-		}, expected: `{"array":[{"a":1},{},{"a":1,"c":3}],"specified":{"a":1,"c":3}}`},
+		}, expectedObject: `{"array":[{"a":1},{},{"a":1,"c":3}],"specified":{"a":1,"c":3}}`,
+			expectedPruned: []string{"specified.b", "array[1].b", "array[2].b", "unspecified"}},
```

Comment:
Not flaky, the test does the sort on the results. I've been assuming that the performance impact of sorting in the unknown field collection step outweighs the benefit of having identical results every time (since the caller can always sort the results themselves if they want it.

It sounds like you'd rather have consistent path ordering everywhere though? Is that true? I can do so throughout if so, jut let me know.

Reply:
API responses should be consistent if possible, and I don't expect there to be significant numbers of unknown fields or pruned paths